plugins/chunter-resources/src/components/CommentInput.svelte

```diff
@@ -17,14 +17,84 @@
   import { Comment } from '@hcengineering/chunter'
   import { Doc, generateId, Ref } from '@hcengineering/core'
   import { getClient } from '@hcengineering/presentation'
+  import { fetchMetadataLocalStorage, setMetadataLocalStorage } from '@hcengineering/ui'
   import { AttachmentRefInput } from '@hcengineering/attachment-resources'
   import { createBacklinks } from '../backlinks'
   import chunter from '../plugin'
 
-  const client = getClient()
   export let object: Doc
+  export let shouldSaveDraft: boolean = true
+
+  const client = getClient()
   const _class = chunter.class.Comment
   let _id: Ref<Comment> = generateId()
+  let inputContent: string = ''
+  let draftComment: Comment | undefined
+
+  $: updateDraft(object)
+
+  $: updateCommentInput(draftComment)
+
+  async function updateCommentInput (draftComment: Comment | undefined) {
+    inputContent = draftComment ? draftComment.message : ''
+  }
+
+  function commentIsEmpty (message: string, attachments: number): boolean {
+    return !(message.length > 0 || attachments > 0)
+  }
+
+  async function updateDraft (object: Doc) {
+    const drafts = fetchMetadataLocalStorage(chunter.metadata.CommentDrafts) ?? {}
```

Comment:
Better to use svelte writable and do fetch only once and save on draft update.

Reply:
Done